### PR TITLE
[13.0] s_packaging_calculator: misc fix/imp

### PR DESCRIPTION
**Pull request by simahawk**
See atomic commits for details.

ref: 2535

### CODE REVIEW

File: stock_packaging_calculator/models/product.py [hunk 1]
```diff
@@ -145,3 +149,57 @@ def _prepare_qty_by_packaging_values(self, packaging, qty_per_pkg):
...
+            return ""
+
+        # Exclude unit qty and reuse it later
+        unit_qty = None
+        has_only_units = True
+        _qty_by_packaging = []
+        for pkg_qty in qty_by_packaging:
+            if pkg_qty["is_unit"]:
+                unit_qty = pkg_qty["qty"]
+                continue
```
Comment (mmequignon) [inline stock_packaging_calculator/models/product.py:175]: what if you have more than 1 packaging with units? Is it even possible?
Shouldn't we set "unit_qty" as 0 and sum the qties instead?

**Review by mmequignon — APPROVED** [review]

File: stock_packaging_calculator/models/product.py (hunk 1 above)
Comment (simahawk) [reply stock_packaging_calculator/models/product.py:175]: impossible. The unit is added explicitly at the end of the pkg list and it's always the product uom.

Comment (mmequignon) [reply stock_packaging_calculator/models/product.py:175]: Okay then

File: stock_packaging_calculator/models/product_qty_by_packaging_mixin.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# @author: Sébastien Alix <sebastien.alix@camptocamp.com>
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl)
+from odoo import api, fields, models
+
+
+class ProductQtyByPackagingMixin(models.AbstractModel):
+    """Allow displaying product qty by packaging.
+    """
+
+    _name = "product.qty_by_packaging.mixin"
```
Comment (TDu) [inline stock_packaging_calculator/models/product_qty_by_packaging_mixin.py:12]: It does not look to me that the new mixin is covered by test, maybe I am mistaken ?

Comment (simahawk) [reply stock_packaging_calculator/models/product_qty_by_packaging_mixin.py:12]: no, I wanted to but I had no time to add fake models. BUT it's tested in #1168 on real models :wink:

Comment (simahawk) [reply stock_packaging_calculator/models/product_qty_by_packaging_mixin.py:12]: @TDu good for you? Then I can merge and rebase #1168

File: stock_packaging_calculator/models/product.py [hunk 3]
```diff
@@ -140,3 +150,57 @@ def _prepare_qty_by_packaging_values(self, packaging, qty_per_pkg):
...
+            has_only_units = False
+            _qty_by_packaging.append(pkg_qty)
+        # Browse them all at once
+        records = self.env["product.packaging"].browse(
+            [x["id"] for x in _qty_by_packaging]
+        )
+        _qty_by_packaging_as_str = self.env.context.get(
+            "_qty_by_packaging_as_str", self._qty_by_packaging_as_str
+        )
```
Comment (sebalix) [inline stock_packaging_calculator/models/product.py:184]: Not related to this PR, I'm wondering if it is a good practice or advisable to pass non-serializable objects into the context?

**Review by sebalix — APPROVED** [review]

**Pull request merged**